pilot/pkg/model/push_context.go

```diff
@@ -2449,7 +2463,13 @@ func (ps *PushContext) mergeGateways(proxy *Proxy) *MergedGateway {
 		return nil
 	}
 
-	return mergeGateways(gatewayInstances, proxy, ps)
+	mgw := mergeGateways(gatewayInstances, proxy, ps)
+
+	if gatewayNames.Len() > 0 {
```

Comment:
Can we add a comment here saying that if `features.EnableStrictGatewayMerging` is false,  then `gatewayNames` will be empty

Reply:
also, if there are more than 1, can we log a warning?